compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java

```diff
@@ -188,9 +192,18 @@ public void genJMethodForBFunc(BIRFunction func, ClassWriter cw, BIRPackage modu
             access += ACC_STATIC;
         }
 
+        indexMap.addIfNotExists(STRAND, symbolTable.stringType);
+
         String funcName = func.name.value;
         BType retType = getReturnType(func);
-        String desc = JvmCodeGenUtil.getMethodDesc(func.type.paramTypes, retType);
+        String desc;
+        int invocationCountArgVarIndex = -1;
+        if (isWorker) {
+            invocationCountArgVarIndex = indexMap.addIfNotExists(INVOCATION_COUNT, symbolTable.stringType);
+            desc = JvmCodeGenUtil.getMethodDescForWorker(func.type.paramTypes, retType);
```

Comment:
As the `getMethodDescForWorker` method is only used here, we can include it in the line like this.
```java 
desc =  INITIAL_METHOD_DESC + "I" + populateMethodDesc(func.type.paramTypes) + generateReturnType(retType);
```